Documentation/tutorial/validate_your_markdown_files.md

```diff
@@ -235,3 +239,106 @@ public class MyInputMetadataValidator : IInputMetadataValidator
 ```
 
 Enable metadata rule is same as other rules, just copy the assemblies to the `plugins` of your template folder and run `docfx`.
+
+### Create configurable metadata validation plug-ins
+
+There are two steps to create a metadata validation:
+
+1.  We need to modify export attribute for metadata validator plug-in:
+
+    ```csharp
+    [Export("hello_is_not_valid", typeof(IInputMetadataValidator))]
+    ```
+
+    > [!Note]
+    > If the rule doesn't have a contract name, it will be always enabled,
+    > i.e., there is no way to disable it unless delete the assembly file.
+
+2.  Modify `md.style` with following content:
+
+    ```json
+    {
+      "metadataRules": [
+        { "contractName": "hello_is_not_valid", "disable": false }
+      ]
+    }
+    ```
+
+## Advanced: Share your rules
+
+Some users have a lot of document projects, and want to share validations for all of them, and don't want to write `md.style` file repeatedly.
+
+### Create template
+For this propose, we can create a template with following structure:
+
+```
+/  (root folder for plug-in)
+\- md.styles
+   |- <category-1>.md.style
+   \- <category-2>.md.style
+\- plugins
+   \- <your_rule>.dll 
+```
+
+In `md.styles` folder, there is a set of definition files, with file extension `.md.style`, each file is a category.
+
+In one category, there is a set of rule definition.
+
+For example, create a file with name `test.md.style`, then write following content:
+
+```json
+{
+   "tagRules": {
+      "heading": {
+         "tagNames": [ "H1", "H2" ],
+         "behavior": "Warning",
+         "messageFormatter": "Please do not use <H1> and <H2>, use '#' and '##' instead.",
+         "openingTagOnly": true
+      }
+   },
+   "rules": {
+      "code": "code_snippet_should_be_csharp"
+   },
+   "metadataRules": {
+     "hello": { "contractName": "hello_is_not_valid", "disable": true }
+   }
+}
+```
+
+Then `test` is the category name (from file name) for three rules, and apply different `id` for each rule, they are `heading`, `rules` and `hello`.
```

Comment:
mistake.

---
In reply to: [74188550](https://github.com/dotnet/docfx/pull/587#discussion_r74188550) [](ancestors = 74188550)